Docblock fixes in TimeInterval

tivnet opened the pull request:
Fixes #
Docblocks only, to please IDEs.
A couple of places `DateTime -> \WC_DateTime`. Others - just `\`

tivnet committed:
Docblock fixes in TimeInterval

adrianduffell reviewed (changes requested):
@tivnet Thanks for contributing this! Overall it looks good.

 I think ideally we should use [`\DateTimeInterface`](https://www.php.net/manual/en/class.datetimeinterface.php) for all of these return types. It would provide flexibility to change the internal date implementation in the future. Does this work ok in your IDE?

tivnet commented:
@adrianduffell 

> Does this work ok in your IDE?

In PhpStorm it works fine:

![image](https://user-images.githubusercontent.com/1696330/111311943-2a89e180-8635-11eb-8e59-f797d0ecc256.png)

Cannot say anything about other IDEs.

adrianduffell commented:
> In PhpStorm, `@return \DateTimeInterface` works fine:

@tivnet Thanks, let's go ahead with changing it to `\DateTimeInterface` and then I think this will be good to merge 🙂

tivnet committed:
TimeInterval - use DateTimeInterface in docblocks

Also: added `@throws`, reformatted the code, changed `bool` to `true` in a couple of methods.

No code changes, only docblocks.

tivnet commented:
@adrianduffell Done.

adrianduffell commented on src/API/Reports/TimeInterval.php:
```diff
@@ -285,21 +304,23 @@ public static function next_hour_start( $datetime, $reversed = false ) {
 
 		if ( $reversed ) {
-			$hours_offset_timestamp --;
+			$hours_offset_timestamp--;
```
Let's keep the space before `--`

adrianduffell commented on src/API/Reports/TimeInterval.php:
```diff
@@ -30,18 +30,21 @@ class TimeInterval {
 	 *
 	 * @param string $datetime_string String representation of local datetime.
-	 * @return DateTime
+	 *
```
A new line was added here.

adrianduffell commented on src/API/Reports/TimeInterval.php:
```diff
@@ -247,6 +260,7 @@ public static function intervals_between( $start_datetime, $end_datetime, $inter
 					$start_datetime = self::next_week_start( $start_datetime );
 					$week_count++;
 				} while ( $start_datetime <= $end_datetime );
+
```
Let's remove the new lines that were added.

adrianduffell commented on src/API/Reports/TimeInterval.php:
```diff
@@ -574,10 +608,12 @@ public static function normalize_between_params( $request, $param_names, $is_dat
+	 * @param string           $param   Parameter name.
+	 *
+	 * @return \WP_Error|true
+	 * @noinspection PhpUnusedParameterInspection
```
Let's remove the @noinspection tag

adrianduffell reviewed (changes requested):
Hi @tivnet,

Thanks for making the `\DateTimeInterface` change. However it looks like some other changes came along with it compared to your original commit. I'm guessing it was caused by auto formatting on save. I've highlighted some of it in the code comments.

Are you able to remove these?

Thank you!

tivnet committed:
Removed spaces before `return` and `@return`; Removed `@noinspection`; Restored space before unary `--`.

tivnet committed:
Removed spaces before `return` (one more left)

tivnet committed:
Reverted extra blank line 456

tivnet commented:
@adrianduffell I changed the IDE settings... Now it removed more blanks :)

adrianduffell commented:
> @adrianduffell I changed the IDE settings... Now it removed more blanks :)

Thanks @tivnet , the removal of the blank lines in the code is a problem too 😅 . Ideally this PR would just be making docblock changes.

tivnet commented:
@adrianduffell Never mind :)
